lib/server.js

```diff
@@ -1590,10 +1106,44 @@ function paginationStream(opts, onRecord, done) {
     }
 }
 
+function getPlacementData(options) {
+    assert.object(options, 'options');
+    assert.object(options.log, 'options.log');
+    assert.object(options.dataDirector, 'options.dataDirector');
+
+    function _getPlacementData(rpc) {
+        var msgid = rpc.requestId();
+        var argv = rpc.argv();
+
+        if (invalidArgs(rpc, argv, GDP_ARGS_SCHEMA)) {
+            return;
+        }
+
+        dtrace['getplacementdata-start'].fire(function () {
+            return ([msgid]);
+        });
+
+        var log = options.log;
+
+        log.debug('getPlacementData: entered');
+
+        dtrace['getplacementdata-done'].fire(function () {
+            return ([msgid]);
+        });
+
+        log.debug('getPlacementData: done');
+
+        rpc.write(options.dataDirector.dataPlacement);
```

Comment:
Does node-fast do anything for us here in terms of very large responses? I'm wondering about the case where this object could potentially be many MiB and should be chunked up, but I'm not sure if that's electric-boray's responsibility or if this write call will do it for us somewhere in node-fast.

Reply:
A good question, and one I really don't know the answer to. ping @kellymclaughlin do you know off hand? I can dig through node-fast docs/source later and see if we don't.  I imagine it's taken care for us, but I don't have anything to back that thought up.

Reply:
I would leave it to node-fast to do the breaking up. If we have to significantly change the data structure and we are suffering due to the size it may make sense to revisit this, but for now node-fast ought to handle it just fine I think.

Reply:
Great, thanks for the explanation.